src/Stratis.SmartContracts.Tests.Common/MockChain/PoWMockChain.cs

```diff
@@ -54,6 +57,26 @@ public void WaitForAllNodesToSync()
             }
         }
 
+        public void MineBlocks(int num)
+        {
+            for (int i = 0; i < num; i++)
+            {
+                int nodeToMineWith = this.height % this.Nodes.Count;
+
+                this.Nodes[nodeToMineWith].MineBlocks(num);
```

Comment:
I think `num` here should be replaced with `i`, or we should increment the height by `num` and remove the loop. Depends on whether the goal is to spread out the mining of `num` blocks across all miners, or have one miner mine `num` blocks.